dfsclient/src/main/java/org/hpccsystems/dfs/client/PartitionProcessor.java

```diff
@@ -260,20 +260,35 @@ else if (fd.getFieldType() == FieldType.STRING)
         // Construct data partition ranges
         //------------------------------------------------------------------------------
         
-        int numTLKRanges = (tlkRecords.size()-1);
-        if (numTLKRanges != this.dataPartitions.length)
+        if (tlkRecords.size() != this.dataPartitions.length)
         {
-            if (numTLKRanges == 1)
-            {
-                // We cannot currently process this type of TLK. Treat it as non-existent TLK 
-                log.warn("No TLK partition provided to partition processor; all partitions will be returned for all filters.");
-                return;
-            }
-            else
+            // This would represent some form of corruption or an issue during the read
+            throw new Exception("Mismatch between # of data partitions and records in TLK: " + tlkRecords.size() + " # partitions: " + this.dataPartitions.length);
+        }
+
+        // Create a max value cap record
+        if (tlkRecords.size() > 0)
```

Comment:
is it ok to fall through if size() == 0 ?

Reply:
Yep